startup/10-detectors.py

```diff
@@ -589,19 +641,26 @@ def stage(self):
                 #DIRECTORY = "/nsls2/xf07bm/data/pb_data"
 
                 filename = 'an_' + str(uuid.uuid4())[:6]
-                self._full_path = os.path.join(DIRECTORY, filename)  # stash for future reference
+                full_path = make_filename(filename)
+                self._full_path = os.path.join(DIRECTORY, full_path)  # stash for future reference
                 print(">>>>>>>>>>>>>>> writing to {}".format(self._full_path))
 
                 self.filepath.put(self._full_path)
-                self.resource_uid = self._reg.register_resource(
-                    'PIZZABOX_AN_FILE_TXT',
-                    DIRECTORY, self._full_path,
-                    {'chunk_size': self.chunk_size})
-
+                self._resource_uid = str(uuid.uuid4())
+                resource = {'spec' : 'PIZZABOX_AN_FILE_TXT',
+                            'root' : DIRECTORY,
+                            'resource_path': full_path,
+                            'resource_kwargs': {'chunk_size': self.chunk_size},
+                            'path_semantics': os.name,
```

Comment:
I believe it should be the following line according to https://github.com/bluesky/ophyd/pull/705/files#diff-80d06b5ac1409f465083a6abb44a9b14R714:
```suggestion
                        'path_semantics': {'posix': 'posix', 'nt': 'windows'}[os.name]}
```